kong/core/reports.lua

```diff
@@ -81,6 +81,9 @@ local function send_report(signal_type, t, host, port)
         end
 
         v = json
+
+      elseif type(v) == "function" then
+        v = v()
```

Comment:
Good! Last thing: we do have tests for the reports - we'd better test this (anyway you'd like, although I feel like spawning the mock UDP server and waiting for its output might be rather cumbersome, maybe a simple Busted spy can do the job of ensuring we called this value when it is a function).

Also: we should target the `release/0.12.0` branch instead of master.

Reply:
My first thought indeed was to use a spy, but because the spy replaces the function with a proxy table we try to encode it. Rather than bang my head against the wall I just inserted into the existing UDP server call.

Reply:
That's easy to take care of with a mocked `type()` which returns "function" when its argument is equal to the spy (it implies reloading the module though) - but that's fine.

Reply:
Oooh, I didn't think this far ahead. Not gonna tweak here but good to file away for future use, thanks!